File: SDK/python/verifio/history.py
from typing import Optional, cast
from .client import VerifioClient
from .types import PaginationOptions, VerificationResult, PaginatedData

class HistoryService:
    def __init__(self, client: VerifioClient):
        self.client = client

    def list(self, options: Optional[PaginationOptions] = None) -> PaginatedData[VerificationResult]:
        page = options.get("page", 1) if options else 1
        limit = options.get("limit", 20) if options else 20

        response_data = self.client.request("GET", f"/history?page={page}&limit={limit}")

        if not response_data.get("success") or not response_data.get("data"):
            raise Exception(response_data.get("error") or "Failed to get history")

        return response_data["data"]

File: SDK/rust/src/history.rs
use reqwest::Method;
use std::sync::Arc;

use crate::client::VerifioClient;
use crate::error::VerifioError;
use crate::types::{PaginatedData, PaginationOptions, VerificationResult};

#[derive(Clone, Debug)]
pub struct HistoryService {
    client: Arc<VerifioClient>,
}

impl HistoryService {
    pub(crate) fn new(client: Arc<VerifioClient>) -> Self {
        Self { client }
    }

    pub async fn list(
        &self,
        options: Option<PaginationOptions>,
    ) -> Result<PaginatedData<VerificationResult>, VerifioError> {
        let page = options.as_ref().and_then(|o| o.page).unwrap_or(1);
        let limit = options.as_ref().and_then(|o| o.limit).unwrap_or(20);

        let path = format!("/history?page={}&limit={}", page, limit);

        self.client.request::<_, ()>(Method::GET, &path, None).await
    }
}
